gen-ai/orchestrator-server/src/main/python/tock-llm-indexing-tools/index_documents.py

```diff
@@ -92,22 +97,29 @@ async def index_documents(args):
                                         '<embeddings_json_config>'
                                         '<vector_store_json_config>'
                                         '<chunks_size>'
+                                        '<ignore_source>'
 
     Returns:
-        The indexing session unique id.
+        The indexing details.
     """
-    # unique date / uuid for each indexing session (stored as metadata)
+
+    start_time = datetime.now()
+    formatted_datetime = start_time.strftime('%Y-%m-%d %H:%M:%S')
     session_uuid = str(uuid4())
-    logging.debug(
-        f"Beginning indexation session {session_uuid} at '{formatted_datetime}'"
-    )
+    logging.debug(f"Beginning indexation session {session_uuid} at '{formatted_datetime}'")
 
     logging.debug(f"Read input CSV file {args['<input_csv>']}")
-    df = pd.read_csv(args['<input_csv>'], delimiter='|', quotechar='"', header=0) # names=['title', 'source', 'text']
+    df = pd.read_csv(args['<input_csv>'], delimiter='|', quotechar='"', header=0)
+    # Filter the DataFrame to only include rows where the 'text' column is not empty or null
+    df_filtered = df[df['text'].notnull() & df['text'].str.strip().ne('')]
     # Prevent NaN value in the 'source' column with a default value 'UNKNOWN', then replace it with None
-    df['source'] = df['source'].fillna('UNKNOWN')
-    df['source'] = df['source'].replace('UNKNOWN', None)
-    loader = DataFrameLoader(df, page_content_column='text')
+    if bool(args['<ignore_source>']):
+        df_filtered['source'] = None
```

Comment:
I don't know if you remember, but we've been discussing the fact that the pdf urls point to Golven's personal folder, and we can't consider that a valid link for end users, since they don't have access to that path. So we've decided to remove them, and consign the pdf documents as unsourced. 

I see two things: 
- Yes, I'm in favor of keeping this information in the metadata.
- We need to modify the code that processes the pdf to have the Google Drive url of the PDF, which can be given/exposed to the end user.

Reply:
Ok so we could add the source as metadata for instance "original_source" (every time so that when it's ignored we still have the pdf filename present in metadata).
Can you juste add this metadata ?

It will then be used when we debug RAG trace to understand were the chunk came from.